client/state/login/actions.js

```diff
@@ -572,3 +576,51 @@ export const logoutUser = redirectTo => ( dispatch, getState ) => {
 			return Promise.reject( error );
 		} );
 };
+
+/**
+ * Retrieves the type of authentication of the account of the specified user.
+ *
+ * @param {String} usernameOrEmail - id of the user
+ * @return {Function} a promise that will resolve once the authentication account type has been retrieved
+ */
+export const getAuthAccountType = usernameOrEmail => dispatch => {
```

Comment:
It seems to me, since you've went with the data layer approach, the only thing should be here now is:
```js
export const getAuthAccountType = usernameOrEmail => ( {
	type: LOGIN_AUTH_ACCOUNT_TYPE_REQUESTING,
	usernameOrEmail,
} );
```

Also, since you went with the data layer approach, you can't get a promise from here, and therefor the tracking can't rely on the promise returned from here.

Reply:
I'm not sure to understand the first part of your comment. Right now this function is responsible for checking whether the username is empty or not (this cannot be performed by the API as it would return a `404` since that information is part of the endpoint's url itself). It also triggers three distinct events.

The last part of your comment I totally agree with. I've struggled quite a bit with that, and I'll update the code as you suggested in another comment.

Reply:
Why we're checking it here and not in the form?